.github/workflows/pypi-release.yml

```diff
@@ -9,29 +9,62 @@ on:
   workflow_dispatch:
 
 jobs:
-  pypi:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest]
-        python-version: ['3.8']
+        os: [ubuntu-20.04, macos-11]
     steps:
-    - uses: actions/checkout@v3
-      with:
-        submodules: "recursive"
-    - uses: actions/setup-python@v4
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: python3 -m pip install --upgrade pybind11 setuptools wheel
-    - name: Compile project
-      run: python3 -m pip install -e .
-    - name: Build distribution package
-      run: python3 setup.py sdist bdist_wheel
-    - name: Publish distribution 📦 to PyPI, if release
-      uses: pypa/gh-action-pypi-publish@release/v1
-      with:
-        user: __token__
-        password: ${{ secrets.pypi_password_farm_ng_core }}
-      if: github.event_name == 'release' && github.event.action == 'published'
+      - uses: actions/checkout@v4
+        with:
+          submodules: "recursive"
+      # NOTE: enable later to build for aaarch64
+      # NOTE: this is not working yet, because of grpcio dependency
+      #- name: Set up QEMU
+      #  if: runner.os == 'Linux'
+      #  uses: docker/setup-qemu-action@v3
+      #  with:
+      #    platforms: all
+      - name: Install dependencies
+        run: python3 -m pip install --upgrade pybind11 setuptools wheel cibuildwheel==2.16.2
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.16.2
+        env:
+          # Disable building PyPy wheels on all platforms
+          CIBW_SKIP: "pp* *-musllinux_*"
+          # configure cibuildwheel to build native archs ('auto'), and some
+          # emulated ones
+          CIBW_ARCHS_LINUX: auto # aarch64
+      - uses: actions/upload-artifact@v3
+        with:
+          path: ./wheelhouse/*.whl
+
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build sdist
+        run: pipx run build --sdist
+      - uses: actions/upload-artifact@v3
+        with:
+          path: dist/*.tar.gz
+
+  upload_pypi:
+    needs: [build_wheels, build_sdist]
+    runs-on: ubuntu-latest
+    #if: github.event_name == 'release' && github.event.action == 'published'
```

Comment:
uncomment this before merging and we have green here: https://github.com/farm-ng/farm-ng-core/actions/runs/6816813675